lightning/src/util/events.rs

```diff
@@ -63,37 +102,12 @@ pub enum Event {
 	PaymentReceived {
 		/// The hash for which the preimage should be handed to the ChannelManager.
 		payment_hash: PaymentHash,
-		/// The preimage to the payment_hash, if the payment hash (and secret) were fetched via
-		/// [`ChannelManager::create_inbound_payment`]. If provided, this can be handed directly to
-		/// [`ChannelManager::claim_funds`].
-		///
-		/// [`ChannelManager::create_inbound_payment`]: crate::ln::channelmanager::ChannelManager::create_inbound_payment
-		/// [`ChannelManager::claim_funds`]: crate::ln::channelmanager::ChannelManager::claim_funds
-		payment_preimage: Option<PaymentPreimage>,
-		/// The "payment secret". This authenticates the sender to the recipient, preventing a
-		/// number of deanonymization attacks during the routing process.
-		/// It is provided here for your reference, however its accuracy is enforced directly by
-		/// [`ChannelManager`] using the values you previously provided to
-		/// [`ChannelManager::create_inbound_payment`] or
-		/// [`ChannelManager::create_inbound_payment_for_hash`].
-		///
-		/// [`ChannelManager`]: crate::ln::channelmanager::ChannelManager
-		/// [`ChannelManager::create_inbound_payment`]: crate::ln::channelmanager::ChannelManager::create_inbound_payment
-		/// [`ChannelManager::create_inbound_payment_for_hash`]: crate::ln::channelmanager::ChannelManager::create_inbound_payment_for_hash
-		payment_secret: PaymentSecret,
 		/// The value, in thousandths of a satoshi, that this payment is for. Note that you must
 		/// compare this to the expected value before accepting the payment (as otherwise you are
 		/// providing proof-of-payment for less than the value you expected!).
 		amt: u64,
-		/// This is the `user_payment_id` which was provided to
-		/// [`ChannelManager::create_inbound_payment_for_hash`] or
-		/// [`ChannelManager::create_inbound_payment`]. It has no meaning inside of LDK and is
-		/// simply copied here. It may be used to correlate PaymentReceived events with invoice
-		/// metadata stored elsewhere.
-		///
-		/// [`ChannelManager::create_inbound_payment`]: crate::ln::channelmanager::ChannelManager::create_inbound_payment
-		/// [`ChannelManager::create_inbound_payment_for_hash`]: crate::ln::channelmanager::ChannelManager::create_inbound_payment_for_hash
-		user_payment_id: u64,
+		/// Information for claiming this received payment.
+		receipt: ReceiveInfo,
```

Comment:
@jkczyz I think these could use better names if you have any ideas :grimacing: `claim_info: ClaimInfo` instead?

Reply:
We use "claim" in a few other places related to HTLCs. Not totally unrelated but might want to avoid here. How about `purpose: PaymentPurpose`?

Reply:
Hmm, not sure if I would 100% get that if I were a user and received a `PaymentReceived { purpose: PaymentPurpose::Invoice / PaymentPurpose::Spontaneous }`. Maybe missing something?

Maybe `ReceiveType::Invoice { preimage, .. } / ReceiveType::Spontaneous(preimage)`?

Reply:
I read it as: the purpose of the received payment is to pay an invoice. How about `PaymentPurpose::InvoicePayment` and `PaymentPurpose::SpontaneousPayment`?

Generally, my rule of thumb is suffixes like `Data`, `Info`, `Details`, `Type`, etc. don't provide any meaningful information when describing a data structure as these are -- by definition -- types (so why say `Type`?) that store data / information (so specifying that is redundant).

Maybe `Type` is ok when trying to distinguish what becomes before it (though, generally sub-typing provides a mechanism for this). But what comes before it here (i.e., `Receive`) isn't really a _thing_. So maybe this is a good argument for making separate `Event` variants for each payment type (`InvoicePaymentReceived` and `SpontaneousPaymentReceived`) rather than putting an enum inside `PaymentReceived`.

Reply:
> PaymentPurpose::InvoicePayment and PaymentPurpose::SpontaneousPayment

Gonna go with this, lemme know if there's an issue